editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -5598,7 +5603,14 @@ void CanvasItemEditorViewport::_create_nodes(Node *parent, Node *child, String &
 		undo_redo->add_undo_method(ed, "live_debug_remove_node", NodePath(String(EditorNode::get_singleton()->get_edited_scene()->get_path_to(parent)) + "/" + new_name));
 	}
 
-	if (Object::cast_to<TouchScreenButton>(child) || Object::cast_to<TextureButton>(child)) {
+	String node_class = child->get_class();
+	if (node_class == "AudioStreamPlayer2D") {
+		undo_redo->add_do_property(child, "stream", audio_stream);
+	} else if (node_class == "Polygon2D") {
+		undo_redo->add_do_property(child, "texture", texture);
+	} else if (node_class == "TouchScreenButton") {
+		undo_redo->add_do_property(child, "texture_normal", texture);
+	} else if (node_class == "TextureButton") {
```

Comment:
A `switch` statement would be better here. Also you can merge the cases with the same properties.